backend/services/theme_service.py

```
@@ -2,90 +2,146 @@
+            x_end = min(img.shape[1], x + w + x_expand)
+
+            # Set the head area to black (0) to protect it from changes
+            mask[y_start:y_end, x_start:x_end] = 0
+            
+            # Encode the mask back to bytes
+            _, mask_bytes = cv2.imencode('.png', mask)
+            return mask_bytes.tobytes()
+            
+        except Exception as e:
+            logger.error(f"Failed to create head mask: {e}", exc_info=True)
+            raise HTTPException(status_code=500, detail="Could not process image to create head mask.")
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)